calamari-common/calamari_common/remote/mon_remote.py

```diff
@@ -170,6 +171,24 @@ def _get_config(cluster_name):
     return config_response
 
 
+def rbd_listing(cluster_handle):
+    """
+    For each pool list the rbd images
+    return a mapping of pool name to rbd images
+    """
+    listing = {}
+    pools = rados_command(cluster_handle, "osd lspools")
+    for pool in pools:
+        name = pool['poolname']
+        result = rbd_command(['ls', '-l', '--format', 'json'], name)
```

Comment:
The problem with doing this is that you're doing IO to the RADOS cluster, in the same path where you're querying its health (i.e. when calling rbd_listing from cluster_status).  If the RADOS cluster is not healthy, then this is going to block, and you're going to stop getting cluster heartbeats sent to Calamari -- at exactly the moment when you need your health monitoring the most.  To see what I mean, try killing your OSDs.

You *could* use --rados-osd-op-timeout to try and cope, but it's still nasty, as you'll encounter a delay equal to your timeout (multiplied by the number of pools that exist, given how this is coded) on every heartbeat while the cluster is in an unhealthy state.  Clearly you would also need to catch its error, and return something that let the caller know that the call failed, as opposed to thinking all the images had disappeared.

You'd probably be better off not doing IO from this path.  Let something higher level do its own polling of "rbd ls", using a different path that isn't going to interfere with the (much more important) health monitoring.